pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -142,6 +143,9 @@ func (w *watcher) retrieveAZ(ctx context.Context, delay time.Duration, retries i
 		}
 		attempts++
 	}
+	if w.config.AvailabilityZone == "" {
+		log.Info("Availability zone not set, proxy will default to not using zone aware routing. To manually override use the --availabilityZone flag.")
```

Comment:
Users don't typically have ability to set flags - we recommend they use kube-inject or auto-injection. Could you instead point to the istio docs about injection, and make sure there is some example or info on how to do it ?

Reply:
We could but I'm not sure we would expect users to set the flag manually because it's not really viable especially in Kube. Maybe we should remove the manual override phrase from the log?

Reply:
Don't remove the manual override. If we have docs for the pilot agent, lets point to it. If not, its high time that we had docs for pilot agent, just like other auto generated docs in istioctl.

Reply:
is this going to repeat on the console over and over? If so, kill this printout. If not, simply stop with "not using zone aware routing".

Reply:
No. This is outside the for loop and indicates that the agent is giving up trying to enable zone aware routing.